src/app/shared/entry.ts

```diff
@@ -209,10 +209,23 @@ export abstract class Entry implements OnInit, OnDestroy {
       if (foundVersion) {
         return foundVersion;
       }
+    } else {
+      return null;
     }
+  }
 
-    // Select newest last_modified version, if it's the same, choose the top
-    return versions.reduce((a, b) => (b.last_modified > a.last_modified ? b : a));
+  selectTag(versions: Array<Tag>, urlVersion: string, defaultVersion: string): Tag {
+    const selectedTag = this.selectVersion(versions, urlVersion, defaultVersion);
+    return selectedTag || versions.reduce((a,b) => b.last_built > a.last_built ? b : a);
+  }
+
+  selectWorkflowVersion(versions: Array<WorkflowVersion>, urlVersion: string, defaultVersion: string) {
+    const selectedTag = this.selectVersion(versions, urlVersion, defaultVersion);
+    if (selectedTag != null) {
```

Comment:
No big deal, but this method and the previous method are doing the same thing, but this one with `if...else` and line 219 "inline".